scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -237,7 +237,9 @@ object Type {
   @ast class Method(paramss: List[List[Term.Param]], tpe: Type) extends Type {
     checkParent(ParentChecks.TypeMethod)
   }
-  @ast class Placeholder(bounds: Bounds) extends Type
+  @ast class WildcardRef extends scala.meta.Ref
+  @ast class Wildcard(ref: WildcardRef, bounds: Bounds) extends Type
```

Comment:
Why do we need `ref` ? Is it only to get the position for `_`?

Reply:
yes. similar to how we have `Pat.Wildcard`.

i am actually thinking of replacing the variant prefix in Placeholder with an `Option[Mod.Covariant | Mod.Contravariant]`, similarly.